app/code/Magento/Sales/Model/Order/Creditmemo/Total/Discount.php

```diff
@@ -26,6 +27,16 @@ public function collect(\Magento\Sales\Model\Order\Creditmemo $creditmemo)
          * basing on how much shipping should be refunded.
          */
         $baseShippingAmount = $this->getBaseShippingAmount($creditmemo);
+
+        /**
+         * If credit memo's shipping amount is set and Order's shipping amount is 0,
+         *      throw exception with differente message
+         */
+        if ( $baseShippingAmount && $order->getBaseShippingAmount() <= 0 ) {
+            throw new \Magento\Framework\Exception\LocalizedException(
+                new \Magento\Framework\Phrase("You can not refund shipping if there is no shipping amount.",[])
```

Comment:
```suggestion
                new \Magento\Framework\Phrase("You can not refund shipping if there is no shipping amount.", [])
```
there should be a space between parameters